### ISSUE TITLE
BUG FOR WHO DO SWAP WITH ANOTHER ASET TO YBX  WITH LARGE PRICE AND MAKE HF, NET APY, TOKEN YEILD GOES UP SO USER CAN DO BORROW MAX AND OTHER USERS CAN USE THE MAXIMUM LOAN WITH SMALL COLLATERAL SO THAT THEY CAN STEAL THE LOAN WITH MOVING LOAN ASSET TO ANOTHER WALLET

### ISSUE THREAD

**Issue by davidsa05** [script3/yieldblox-beta#317]
User can make max borrow again after someone plays with the price of YBX by swapping at a high price.

screenshoot my account before someone plays YBX price :
![screencapture-ybx-beta-script3-io-2021-10-14-11_26_29](https://user-images.githubusercontent.com/91865890/137319375-4edb5bf9-850f-44c5-8d86-f8eabb400613.png)
![screencapture-ybx-beta-script3-io-2021-10-14-11_26_07](https://user-images.githubusercontent.com/91865890/137319593-ff34f151-c420-447a-b9cd-5b2034c80820.png)

screenshoot my account after someone plays YBX price :
![screencapture-ybx-beta-script3-io-2021-10-14-19_30_01](https://user-images.githubusercontent.com/91865890/137319546-cb558b8d-2e34-4c56-b88c-bbe14ab7a8aa.png)
![screencapture-ybx-beta-script3-io-2021-10-14-19_30_12](https://user-images.githubusercontent.com/91865890/137319653-d19596f2-8d88-413c-baa2-dba77647e8e1.png)

And I try to make some borrow, 
![screencapture-ybx-beta-script3-io-2021-10-14-19_30_32](https://user-images.githubusercontent.com/91865890/137319778-8e1d4ff0-0fc4-4cb7-b0f3-f2a6d5b03b78.png)

![14 2](https://user-images.githubusercontent.com/91865890/137319814-41f64640-58b5-4934-a5c9-6d7eff015620.JPG)

and the result is a success 

![14 3](https://user-images.githubusercontent.com/91865890/137319951-47680514-40e1-477c-a248-53ce4cf3745a.JPG)
![screencapture-ybx-beta-script3-io-2021-10-14-19_46_48](https://user-images.githubusercontent.com/91865890/137320162-ad4824de-d213-4c9d-bfd4-8e8a793e4245.png)

**Comment by davidsa05** [comment]
> User can make max borrow again after someone plays with the price of YBX by swapping at a high price.
> 
> screenshoot my account before someone plays YBX price : ![screencapture-ybx-beta-script3-io-2021-10-14-11_26_29](https://user-images.githubusercontent.com/91865890/137319375-4edb5bf9-850f-44c5-8d86-f8eabb400613.png) ![screencapture-ybx-beta-script3-io-2021-10-14-11_26_07](https://user-images.githubusercontent.com/91865890/137319593-ff34f151-c420-447a-b9cd-5b2034c80820.png)
> 
> screenshoot my account after someone plays YBX price : ![screencapture-ybx-beta-script3-io-2021-10-14-19_30_01](https://user-images.githubusercontent.com/91865890/137319546-cb558b8d-2e34-4c56-b88c-bbe14ab7a8aa.png) ![screencapture-ybx-beta-script3-io-2021-10-14-19_30_12](https://user-images.githubusercontent.com/91865890/137319653-d19596f2-8d88-413c-baa2-dba77647e8e1.png)
> 
> And I try to make some borrow, ![screencapture-ybx-beta-script3-io-2021-10-14-19_30_32](https://user-images.githubusercontent.com/91865890/137319778-8e1d4ff0-0fc4-4cb7-b0f3-f2a6d5b03b78.png)
> 
> ![14 2](https://user-images.githubusercontent.com/91865890/137319814-41f64640-58b5-4934-a5c9-6d7eff015620.JPG)
> 
> and the result is a success
> 
> ![14 3](https://user-images.githubusercontent.com/91865890/137319951-47680514-40e1-477c-a248-53ce4cf3745a.JPG) ![screencapture-ybx-beta-script3-io-2021-10-14-19_46_48](https://user-images.githubusercontent.com/91865890/137320162-ad4824de-d213-4c9d-bfd4-8e8a793e4245.png)

it allows people to manipulate the price of YBX if the system can't recognize unfair transactions so that users with a small amount of capital can add more loans to the maximum and can make them get bigger returns in farming YBX.

you have to set limits/parameter so that system can recognize abnormal transactions related to YBX price manipulation so that it can prevent very high spikes related to HF, Net APY, And Token Yeild.

**Comment by davidsa05** [comment]
relate #313 @mootz12 I have an opinion, if the YBX at stake wasn't used as collateral then this bug probably wouldn't have happened if anyone manipulated the price

Connecting this recent bug issue is highly correlated. So if there is a user who has, for example: 10 YBX and he only has 1$ capital to be used as collateral. he does stake 10. this allows him to borrow more if the price of YBX rises very high with manipulation or someone incorrectly inputs the purchase price of YBX with a very large capital

**Comment by mootz12** [comment]
Yes! YBX is at a very limited LTV ratio. (40%). That means, you can borrow against your staked YBX for 40% of its current value.

Due to the volatility that will likely be present, this is INCREDIBLY RISKY. It's up the user who borrows against their YBX to make the appropriate judgement.

Per the buy wall issue on the Testnet, we attempted to dis-incentivize users to trade for YBX. However, we still need paths to allow for our protocol to buy and sell YBX when it needs to. So, we set some really high price barriers. Since a trade occurred, it threw off the TWAP price of YBX by a LOT, since these trades are so out of line. This disparity is impossible on the mainnet, as real markets can't function like this. 

However, on the mainnet, a user can buy a whole bunch of YBX to manipulate the price, stake it all, then borrow money from the pool for ~40% of the YBX's worth. As the price returns to normal, this user will get liquidated, and the pool will sell YBX in exchange for the asset they borrowed. This will result in a net loss for the manipulator, because they will be buying YBX at a high price to push the market, and the pool will be selling it on their behalf at a lower one.

The lower the LTV, the less risky this situation can be. Governance can be used in the future to shift the LTV ratio of YBX to ensure the pool remains healthy.

**Comment by davidsa05** [comment]
Thanks for the explanation @mootz12 , this is very interesting.
But if it works on the mainnet, what I really mean is not the perpetrator who manipulates the price, but if there is a very rich person who wants to buy YBX with large capital, with his ignorance/understanding, he entered the price wrong and made the price go up high , automatically the health factor also goes up.

Before the YBX price returns to normal, there must be one or two people who are lucky to be online at that time and it turns out that his account only has 100 YBX capital to stake and 2$ to lend, but he can take advantage of the moment by borrowing more to get YBX's more farming results, or worse, his loan results can be withdrawn to another wallet, with less asset collateral, so he still gets a profit.

For example, with yesterday's incident, there were people who didn't know anything about swapping assets to YBX, there must be some other users who took advantage of this bug to borrow more max.

Can liquidation compensate for the loss of the protocol system even though the capital he has guaranteed is smaller than what he borrowed?

Could this fear of mine come true? Can users disburse loan assets to other wallets? is there a system that can freeze the wallet if one is exposed to liquid?

**Comment by yuminachan** [comment]
I hope this issue will not happen on mainnet.

$PB (PancakeBunny) also ever got manipulate price recently and the manipulator successful to stole over $200M loan. 
Link news : https://coinmarketcap.com/alexandria/article/pancake-bunny-price-crashes-after-defi-token-targeted-in-flash-loan-attack
Official report: https://pancakebunny.medium.com/hello-bunny-fam-a7bf0c7a07ba
Simulation tragedy: https://medium.com/amber-group/bsc-flash-loan-attack-pancakebunny-3361b6d814fd

We cannot say "Impossible" if there is still have "1% Chance" to make this issue come true. even you also know XLM sometimes ago get dumped so hard and jumped so hard because of bot bugs. while on yesterday's issue, the bugs almost happen in XLM-USDC pair not on XLM-YBX pair which is jumping the price from 0.3x to 1000USDC per XLM.

So, i hope you have made prevented action before this kind of issue happened. Maybe like make an algorithm with the trigger if unusual change price happens to YBX, then protocol gives auto-protect by making the maximum loan reduced significant or making YBX collateral rate (LTV) reduced until several times or freeze allocation to borrow until the manual check is done by the dev to avoid exploits be like this happen again. Just share my thought

**Comment by davidsa05** [comment]
@mootz12 @markuspluna can you follow up this case too?
